lib/src/config_value_factory.cc

```diff
@@ -0,0 +1,63 @@
+#include <hocon/config_value_factory.hpp>
+#include <internal/values/config_null.hpp>
+#include <internal/values/config_string.hpp>
+#include <internal/values/config_long.hpp>
+#include <internal/values/config_double.hpp>
+#include <internal/values/config_int.hpp>
+#include <internal/values/config_boolean.hpp>
+#include <internal/values/simple_config_list.hpp>
+#include <internal/values/simple_config_object.hpp>
+
+namespace hocon {
+
+    using namespace std;
+
+    class config_value_visitor : public boost::static_visitor<shared_value> {
+    public:
+        shared_value operator()(boost::blank null_value) const {
+            return make_shared<const config_null>(nullptr);
+        }
+
+        shared_value operator()(string str) const {
+            return make_shared<const config_string>(nullptr, str, config_string_type::QUOTED);
+        }
+
+        // TODO: Java passes null for original text, does this do the right equivalent thing?
```

Comment:
The place to look is where we use that value, aka https://github.com/puppetlabs/cpp-hocon/blob/master/lib/src/values/config_long.cc#L10-L17.